.github/workflows/cibuild.yml

```diff
@@ -0,0 +1,67 @@
+name: CI Build
+on:
+  workflow_call:
+    inputs:
+      maven_version:
+        description: Maven version
+        default: "3.9.5"
+        required: false
+        type: string
+      maven_profiles:
+        description: Maven Profiles
+        default: "exo-release"
+        required: false
+        type: string
+      extra_maven_opts:
+        description: Extra Maven OPTS
+        default: ""
+        required: false
+        type: string
+      jdk_major_version:
+        description: JDK Major version (eg 8, 11, 14, 17, 21,...)
+        default: 17
+        required: false
+        type: number
+      jdk_distribution:
+        description: OpenJDK Adopted Distribution (temurin, zulu, adopt, liberica,...)
+        default: "temurin"
+        required: false
+        type: string
+    secrets:
+      NEXUS_USERNAME:
+        required: true
+      NEXUS_PASSWORD:
+        required: true
+env: 
+  SETTINGS_XML_URL: 'http://storage.exoplatform.org/public/githubci/maven-settings.xml'
+jobs:
+  build-ci:
+    name: CI Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK ${{ inputs.jdk_major_version }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ inputs.jdk_major_version }}
+          distribution: ${{ inputs.jdk_distribution }}
+      - name: Set up Maven ${{ inputs.maven_version }}
+        uses: stCarolas/setup-maven@v4.5
+        with:
+          maven-version: ${{ inputs.maven_version }}
+      - name: Cache Maven packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Prepare environment 
+        run: |
+          wget -q "${SETTINGS_XML_URL}" -O /tmp/settings.xml
```

Comment:
if put in .m2, no need to the s option I think

Reply:
Yes, but, we wouldn't need to cache this file by the previous step

Reply:
Is there a constraint to put il in the .m2 dirctly instead of /tmp ? If no constraint, you may put the file in the .m2 folder and suppress the -s in this line: https://github.com/exoplatform/swf-scripts/pull/120/commits/a308a9b8069cb2bd94238c2b604dc24d4ca9f4b3#diff-c0686c29bac861bc2f93c992fc7f444c69920f526bc26f49d0dec78384b7ca15R67

Reply:
Updated